Week1/homework/JS-Exercises/Exercise-08-Type checker.js

```diff
@@ -0,0 +1,30 @@
+let firstItem = "First String"; secondItem = "Second String";
+let thirdItem = {
+        name : "First Object",
+        number : 1,
+                };
+    fourthItem = {
+        name : "Second Object",
+        number : 2,
+                 };
+
+typeof(firstItem) == typeof(secondItem) ? console.log("firstItem and secondItem are same type.") : console.log("firstItem and secondItem are not the same.");
+
+typeof(firstItem) == typeof(thirdItem) ? console.log("firstItem and thirdItem are same type.") : console.log("firstItem and thirdItem are not the same.");
+
+typeof(firstItem) == typeof(fourthItem) ? console.log("firstItem and fourthItem are same type.") : console.log("firstItem and fourthItem are not the same.");
+
+typeof(secondItem) == typeof(thirdItem) ? console.log("secondItem and thirdItem are same type.") : console.log("secondItem and thirdItem are not the same.");
+
+typeof(secondItem) == typeof(fourthItem) ? console.log("secondItem and fourthItem are same type.") : console.log("secondItem and fourthItem are not the same.");
+
+typeof(thirdItem) == typeof(fourthItem) ? console.log("thirdItem and fourthItem are same type.") : console.log("thirdItem and fourthItem are not the same.");
+
+typeof(thirdItem.name) == typeof(fourthItem.name) ? console.log("name of thirdItem and name of fourthItem are same type.") : console.log("name of thirdItem and name of fourthItem are not the same.");
+
+typeof(thirdItem.number) == typeof(fourthItem.number) ? console.log("number of thirdItem and number of fourthItem are same type.") : console.log("number of thirdItem and number of fourthItem are not the same.");
+
+
+console.log("The type of firstItem is " + typeof(firstItem) + ".");
+
+console.log("The type of thirdItem is " + typeof(thirdItem) + ".");
```

Comment:
Ah. Few things here:
1. You could use `const` instead of `let` for creating these variables since they aren't being reassigned.
2. There is some indentation problem when you are creating `thirdItem` variable.
3. You didn't use any keyword `let` or `const` for creating the `fourthItem` variable.
4. You should always strive to use triple equals(===) instead of double equals(==) for comparing in JavaScript. This is because triple equals is safer than the double equals since it not only checks for values equality but also that their types are same.
5. I don't think this is the right use of conditional operators(`? :`). We should, ideally, use them during as assignment operation rather than providing a statement in it. A simple `if-else` block would be a better use-case here. 🙂